fenicsprecice/fenicsprecice.py

```diff
@@ -273,9 +286,47 @@ def initialize(self, coupling_subdomain, mesh, function_space, write_function=No
             Recommended time step value from preCICE.
         """
 
-        coords = function_space.tabulate_dof_coordinates()
+        write_function_space, write_function = None, None
+        if type(write_object) is Function:
+            write_function_space = write_object.function_space()
+            write_function = write_object
+        elif type(write_object) is FunctionSpace:
+            write_function_space = write_object
+            write_function = None
+        elif write_object is None:
+            print("Participant {} is read-only participant".format(self._config.get_participant_name()))
+        else:
+            raise Exception("Given write object is neither of type dolfin.functions.function.Function or "
+                            "dolfin.functions.functionspace.FunctionSpace")
+
+        if read_function_space is None and write_function_space:
+            self._coupling_type = CouplingMode.UNIDIR_WRITE
+            assert(self._config.get_write_data_name())
+            print("Participant {} is write-only participant".format(self._config.get_participant_name()))
+        elif read_function_space and write_function_space is None:
+            self._coupling_type = CouplingMode.UNIDIR_READ
+            assert(self._config.get_read_data_name())
+        elif read_function_space and write_function_space:
+            self._coupling_type = CouplingMode.BIDIR
+            assert(self._config.get_read_data_name() and self._config.get_write_data_name())
+        else:
+            raise Exception("Incorrect read and write function space combination provided. Please check input "
+                            "parameters in initialization")
```

Comment:
```suggestion
        elif read_function_space is None and write_function_space is None:
            raise Exception("You did not provide a read_function_space or write_function_space. "
                                      "Please provide a write_function_space, for one-way coupling, where this "
                                      "participant only writes. Please provide a read_function_space, for one-way "
                                      "coupling, where this participant only reads. If you want to realize two-way"
                                      "coupling provide both, read_function_space and write_function_space.")            
        else:
            raise Exception("Incorrect read and write function space combination provided. Please check input "
                            "parameters in initialization")
```
Again, might be a bit verbose, but we should check for this specific case and give a specific error. The `else:` is for everything that we do not expect.